### PR TITLE
ddl : support create sequence in TiDB. 

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve? 
support create sequence in TiDB.
Linked Parser: [#691](https://github.com/pingcap/parser/pull/691)
```
*      CREATE [TEMPORARY] SEQUENCE [IF NOT EXISTS] sequence_name
*      [ INCREMENT [ BY | = ] increment ]
*      [ MINVALUE [=] minvalue | NO MINVALUE | NOMINVALUE ]
*      [ MAXVALUE [=] maxvalue | NO MAXVALUE | NOMAXVALUE ]
*      [ START [ WITH | = ] start ]
*      [ CACHE [=] cache | NOCACHE | NO CACHE]
*      [ CYCLE | NOCYCLE | NO CYCLE]
*      [ ORDER | NOORDER | NO ORDER]
*      [table_options]
```

### What is changed and how it works?
Add sequence.go in ddl package.
Add create sequence interface in ddl api.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to update the documentation

Release note

 - support create sequence in TiDB.

### CODE REVIEW

File: ddl/sequence.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
+// Copyright 2015 PingCAP, Inc.
```
Comment (wjhuang2016) [inline ddl/sequence.go]: 2015 -> 2019

File: ddl/ddl.go [hunk 2]
```diff
@@ -226,6 +226,19 @@ var (
 	ErrFieldNotFoundPart = terror.ClassDDL.New(mysql.ErrFieldNotFoundPart, mysql.MySQLErrName[mysql.ErrFieldNotFoundPart])
 	// ErrWrongTypeColumnValue returns 'Partition column values of incorrect type'
 	ErrWrongTypeColumnValue = terror.ClassDDL.New(mysql.ErrWrongTypeColumnValue, mysql.MySQLErrName[mysql.ErrWrongTypeColumnValue])
+
+	// ErrSequenceRunOut returns when the sequence has been run out.
+	ErrSequenceRunOut = terror.ClassDDL.New(mysql.ErrSequenceRunOut, mysql.MySQLErrName[mysql.ErrSequenceRunOut])
+	// ErrSequenceInvalidData returns when sequence values are conflicting.
+	ErrSequenceInvalidData = terror.ClassDDL.New(mysql.ErrSequenceInvalidData, mysql.MySQLErrName[mysql.ErrSequenceInvalidData])
```
Comment (reafans) [inline ddl/ddl.go]:
```suggestion
	ErrInvalidSequenceData = terror.ClassDDL.New(mysql.ErrInvalidSequenceData, mysql.MySQLErrName[mysql.ErrInvalidSequenceData])
```

File: ddl/db_test.go [hunk 3]
```diff
@@ -4230,6 +4230,57 @@ func (s *testDBSuite2) TestDDLWithInvalidTableInfo(c *C) {
...
+	c.Assert(sequenceTable.Meta().Sequence.Increment, Equals, model.DefaultSequenceIncrementValue)
+	c.Assert(sequenceTable.Meta().Sequence.Start, Equals, model.DefaultPositiveSequenceStartValue)
+	c.Assert(sequenceTable.Meta().Sequence.MinValue, Equals, model.DefaultPositiveSequenceMinValue)
+	c.Assert(sequenceTable.Meta().Sequence.MaxValue, Equals, model.DefaultPositiveSequenceMaxValue)
+	c.Assert(sequenceTable.Meta().Sequence.Cache, Equals, true)
+	c.Assert(sequenceTable.Meta().Sequence.CacheValue, Equals, model.DefaultSequenceCacheValue)
+	c.Assert(sequenceTable.Meta().Sequence.Cycle, Equals, false)
+	c.Assert(sequenceTable.Meta().Sequence.Order, Equals, false)
+}
+
```
Comment (djshow832) [inline ddl/db_test.go]: How about privileges?

File: ddl/db_test.go [hunk 4]
```diff
@@ -4230,6 +4230,57 @@ func (s *testDBSuite2) TestDDLWithInvalidTableInfo(c *C) {
...
+	sequenceTable := testGetTableByName(c, s.s, "test", "seq")
+	c.Assert(sequenceTable.Meta().IsSequence(), Equals, true)
+	c.Assert(sequenceTable.Meta().Sequence.Increment, Equals, model.DefaultSequenceIncrementValue)
+	c.Assert(sequenceTable.Meta().Sequence.Start, Equals, model.DefaultPositiveSequenceStartValue)
+	c.Assert(sequenceTable.Meta().Sequence.MinValue, Equals, model.DefaultPositiveSequenceMinValue)
+	c.Assert(sequenceTable.Meta().Sequence.MaxValue, Equals, model.DefaultPositiveSequenceMaxValue)
+	c.Assert(sequenceTable.Meta().Sequence.Cache, Equals, true)
+	c.Assert(sequenceTable.Meta().Sequence.CacheValue, Equals, model.DefaultSequenceCacheValue)
+	c.Assert(sequenceTable.Meta().Sequence.Cycle, Equals, false)
+	c.Assert(sequenceTable.Meta().Sequence.Order, Equals, false)
```
Comment (djshow832) [inline ddl/db_test.go]: Add more test cases to test these options.

File: ddl/ddl.go [hunk 5]
```diff
@@ -226,6 +226,19 @@ var (
...
+	// ErrSequenceAccessFail returns when sequences are not able to access.
+	ErrSequenceAccessFail = terror.ClassDDL.New(mysql.ErrSequenceAccessFail, mysql.MySQLErrName[mysql.ErrSequenceAccessFail])
+	// ErrNotSequence returns when object is not a sequence.
+	ErrNotSequence = terror.ClassDDL.New(mysql.ErrNotSequence, mysql.MySQLErrName[mysql.ErrNotSequence])
+	// ErrUnknownSequence returns when drop / alter unknown sequence.
+	ErrUnknownSequence = terror.ClassDDL.New(mysql.ErrUnknownSequence, mysql.MySQLErrName[mysql.ErrUnknownSequence])
+	// ErrSequenceUnsupportedTableOption returns when unsupported table option exists in sequence.
+	ErrSequenceUnsupportedTableOption = terror.ClassDDL.New(mysql.ErrSequenceUnsupportedTableOption, mysql.MySQLErrName[mysql.ErrSequenceUnsupportedTableOption])
```
Comment (djshow832) [inline ddl/ddl.go]: Are these errors used in this PR? If they are, add test cases for them.

Comment (AilinKid) [reply ddl/ddl.go]: Some errors are prepared for other sequence operations.

**Comment by wjhuang2016** [comment]
No rollback logic?

**Comment by AilinKid** [comment]
> No rollback logic?

Em, forgot, I know

File: ddl/sequence_test.go [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+func (s *testSequenceSuite) TestCreateSequence(c *C) {
+	s.tk = testkit.NewTestKit(c, s.store)
+	s.tk.MustExec("use test")
+	s.tk.MustGetErrCode("create sequence `seq  `", mysql.ErrWrongTableName)
+
+	// maxvalue should be larger than minvalue.
+	_, err := s.tk.Exec("create sequence seq maxvalue 1 minvalue 2")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[ddl:4136]Sequence 'test.seq' values are conflicting")
```
Comment (tangenta) [inline ddl/sequence_test.go]: Why not using MustGetErrCode?

File: ddl/sequence_test.go [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+	_, err = s.tk.Exec("create sequence seq maxvalue 1 minvalue 1")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[ddl:4136]Sequence 'test.seq' values are conflicting")
+
+	// maxvalue shouldn't be equal to MaxInt64.
+	_, err = s.tk.Exec("create sequence seq maxvalue 9223372036854775807 minvalue 1")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[ddl:4136]Sequence 'test.seq' values are conflicting")
+
+	// TODO : minvalue shouldn't be equal to MinInt64.
```
Comment (tangenta) [inline ddl/sequence_test.go]: Why not implement it?

File: meta/meta.go [hunk 8]
```diff
@@ -181,6 +186,26 @@ func (m *Meta) GetAutoTableID(dbID int64, tableID int64) (int64, error) {
...
+	// Check if DB exists.
+	dbKey := m.dbKey(dbID)
+	if err := m.checkDBExists(dbKey); err != nil {
+		return 0, errors.Trace(err)
+	}
+	// Check if sequence exists.
+	tableKey := m.tableKey(sequenceID)
+	if err := m.checkTableExists(dbKey, tableKey); err != nil {
+		return 0, errors.Trace(err)
+	}
```
Comment (zimulala) [inline meta/meta.go]: The code of line191-200 is the same as line170-180. Could we extract these codes into a function?

File: meta/meta.go [hunk 9]
```diff
@@ -181,6 +186,26 @@ func (m *Meta) GetAutoTableID(dbID int64, tableID int64) (int64, error) {
...
+	return m.txn.HInc(dbKey, m.sequenceKey(sequenceID), step)
+}
+
+// GetSequenceID gets current sequence value with sequence id.
+func (m *Meta) GetSequenceID(dbID int64, sequenceID int64) (int64, error) {
```
Comment (zimulala) [inline meta/meta.go]: This function does the same work as `GetAutoTableID`. Could we remove it?

File: ddl/sequence_test.go (hunk 6 above)
Comment (AilinKid) [reply ddl/sequence_test.go]: Em, the error defined in DDL package will all get 1105.

File: ddl/sequence_test.go (hunk 7 above)
Comment (AilinKid) [reply ddl/sequence_test.go]: Int64 from `-9223372036854775808 ~ 9223372036854775807`
by now, parser parse a negative number use the rule ` '-' NUM ` and action like `$$ = -$2.(int64)`, so here is a mistake, `$2` won't get to the abs(-9223372036854775808).

by the way, do you have some good ideas about this?

File: ddl/sequence.go [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+		case ast.SequenceMinValue:
+			sequenceInfo.MinValue = op.IntValue
+			minSetFlag = true
+		case ast.SequenceMaxValue:
+			sequenceInfo.MaxValue = op.IntValue
+			maxSetFlag = true
+		case ast.SequenceCache:
+			sequenceInfo.CacheValue = op.IntValue
+		case ast.SequenceNoCache:
+			sequenceInfo.Cache = false
```
Comment (wjhuang2016) [inline ddl/sequence.go]: Can we ignore these options since it's false by default?

File: ddl/sequence_test.go (hunk 6 above)
Comment (bb7133) [reply ddl/sequence_test.go]: This is not the case, the error code 1105 just because it is not initialized properly.

File: ddl/sequence.go [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+		case ast.SequenceCycle:
+			sequenceInfo.Cycle = true
+		case ast.SequenceNoCycle:
+			sequenceInfo.Cycle = false
+		case ast.SequenceOrder:
```
Comment (bb7133) [inline ddl/sequence.go]: Now I don't think we can support Order option, maybe we should remove it.

File: ddl/sequence_test.go (hunk 6 above)
Comment (AilinKid) [reply ddl/sequence_test.go]: Oh, maybe I know what's wrong!

File: ddl/ddl_api.go [hunk 12]
```diff
@@ -4021,3 +4021,46 @@ func (d *ddl) RepairTable(ctx sessionctx.Context, table *ast.TableName, createSt
...
+	if !ok {
+		return infoschema.ErrDatabaseNotExists.GenWithStackByArgs(ident.Schema)
+	}
+	_, err := is.TableByName(ident.Schema, ident.Name)
+	if err == nil {
+		if stmt.IfNotExists {
+			ctx.GetSessionVars().StmtCtx.AppendNote(infoschema.ErrTableExists.GenWithStackByArgs(ident))
+			return nil
+		}
+		return infoschema.ErrTableExists.GenWithStackByArgs(ident)
```
[the remark this answers is not in the document]
Comment (AilinKid) [reply ddl/ddl_api.go]: yes, mariadb seems to do the same

File: ddl/ddl_api.go [hunk 13]
```diff
@@ -4021,3 +4021,46 @@ func (d *ddl) RepairTable(ctx sessionctx.Context, table *ast.TableName, createSt
...
+	}
+	if err = checkTooLongTable(ident.Name); err != nil {
+		return err
+	}
+	sequenceInfo, err := buildSequenceInfo(stmt, ident)
+	if err != nil {
+		return err
+	}
+
+	tbInfo, err := buildTableInfo(ctx, d, ident.Name, nil, nil)
```
[the remark this answers is not in the document]
Comment (AilinKid) [reply ddl/ddl_api.go]: Addressed.

File: ddl/ddl_api.go [hunk 14]
```diff
@@ -4021,3 +4021,46 @@ func (d *ddl) RepairTable(ctx sessionctx.Context, table *ast.TableName, createSt
...
+		SchemaID:   schema.ID,
+		TableID:    tbInfo.ID,
+		SchemaName: schema.Name.L,
+		Type:       model.ActionCreateSequence,
+		BinlogInfo: &model.HistoryInfo{},
+		Args:       []interface{}{tbInfo},
+	}
+
+	err = d.doDDLJob(ctx, job)
+	err = d.callHookOnChanged(err)
```
[the remark this answers is not in the document]
Comment (AilinKid) [reply ddl/ddl_api.go]: yes, see line L4034

File: ddl/sequence.go (hunk 10 above)
Comment (AilinKid) [reply ddl/sequence.go]: yes, we could.

File: ddl/ddl_api.go (hunk 14 above)
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply ddl/ddl_api.go]: But we may get alreadyExistErr in onCreateSequence

[the remark this answers is not in the document]
Comment (AilinKid) [reply ddl/ddl_api.go]: you mean the sequence will existence in ddl_worker under the condition of schema change, and finally, it gets an alreadyExistErr here.

ok, got it.

[the remark this answers is not in the document]
Comment (wjhuang2016) [reply ddl/ddl_api.go]: If we create the second sequence after L4034, the first create sequence may get error in `onCreateSequence`.

File: ddl/sequence.go (hunk 11 above)
Comment (AilinKid) [reply ddl/sequence.go]: addressed.

File: ddl/ddl_api.go [hunk 15]
```diff
@@ -4021,3 +4021,46 @@ func (d *ddl) RepairTable(ctx sessionctx.Context, table *ast.TableName, createSt
...
+
+func (d *ddl) CreateSequence(ctx sessionctx.Context, stmt *ast.CreateSequenceStmt) error {
+	ident := ast.Ident{Name: stmt.Name.Name, Schema: stmt.Name.Schema}
+	is := d.GetInfoSchemaWithInterceptor(ctx)
+	schema, ok := is.SchemaByName(ident.Schema)
+	if !ok {
+		return infoschema.ErrDatabaseNotExists.GenWithStackByArgs(ident.Schema)
+	}
+	_, err := is.TableByName(ident.Schema, ident.Name)
+	if err == nil {
```
[the remark this answers is not in the document]
Comment (AilinKid) [reply ddl/ddl_api.go]: addressed.

File: ddl/ddl_api.go (hunk 14 above)
[the remark this answers is not in the document]
Comment (AilinKid) [reply ddl/ddl_api.go]: Share the same point, I see.

File: ddl/sequence.go (hunk 11 above)
Comment (AilinKid) [reply ddl/sequence.go]: addressed.

File: meta/meta.go (hunk 9 above)
Comment (AilinKid) [reply meta/meta.go]: it's a mistake, addressed.

File: ddl/ddl.go (hunk 2 above)
Comment (AilinKid) [reply ddl/ddl.go]: nice, I will take care of it next pr.

File: ddl/sequence_test.go (hunk 6 above)
Comment (AilinKid) [reply ddl/sequence_test.go]: Addressed.

File: ddl/sequence_test.go (hunk 7 above)
Comment (tangenta) [reply ddl/sequence_test.go]: Get it. Later I will fix the parser.

File: ddl/sequence.go [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+		case ast.SequenceMaxValue:
+			sequenceInfo.MaxValue = op.IntValue
+			maxSetFlag = true
+		case ast.SequenceCache:
+			sequenceInfo.CacheValue = op.IntValue
+		case ast.SequenceNoCache:
+			sequenceInfo.Cache = false
+		case ast.SequenceCycle:
+			sequenceInfo.Cycle = true
+		case ast.SequenceOrder:
```
Comment (wjhuang2016) [inline ddl/sequence.go]: Why is `case ast.SequenceOrder` still here?

File: ddl/sequence.go [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+			// TODO : TiDB don't actually support ORDER option by now.
+		case ast.SequenceNoCycle, ast.SequenceNoOrder:
+		}
+	}
+	// Fill the default value, min/max/start should be adjusted with increment's positive and negative.
+	if !minSetFlag || !maxSetFlag || !startSetFlag {
```
Comment (wjhuang2016) [inline ddl/sequence.go]: Can we use de' morgan law to change the logic?

File: ddl/ddl_api.go [hunk 18]
```diff
@@ -4139,3 +4139,54 @@ func (d *ddl) RepairTable(ctx sessionctx.Context, table *ast.TableName, createSt
...
+		SchemaID:   schema.ID,
+		TableID:    tbInfo.ID,
+		SchemaName: schema.Name.L,
+		Type:       model.ActionCreateSequence,
+		BinlogInfo: &model.HistoryInfo{},
+		Args:       []interface{}{tbInfo},
+	}
+
+	err = d.doDDLJob(ctx, job)
+	// sequence exists, but if_not_exists flags is true, so it ignore this error.
```
[the remark this answers is not in the document]
Comment (AilinKid) [reply ddl/ddl_api.go]: Addressed.

File: ddl/sequence.go (hunk 17 above)
Comment (AilinKid) [reply ddl/sequence.go]: nice idea, addressed.

File: ddl/sequence.go (hunk 16 above)
Comment (AilinKid) [reply ddl/sequence.go]: addressed.

File: ddl/ddl_api.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply ddl/ddl_api.go]: I still don't think it's appropriate， please add a TODO about “refine the error”.

File: ddl/sequence.go [hunk 19]
```diff
@@ -0,0 +1,178 @@
...
+			maxSetFlag = true
+		case ast.SequenceCache:
+			sequenceInfo.CacheValue = op.IntValue
+		case ast.SequenceNoCache:
+			sequenceInfo.Cache = false
+		case ast.SequenceCycle:
+			sequenceInfo.Cycle = true
+		case ast.SequenceNoCycle:
+			sequenceInfo.Cycle = false
+			// TODO: TiDB haven't actually support ORDER option by now.
```
Comment (wjhuang2016) [inline ddl/sequence.go]: Remove the `TODO`

File: ddl/sequence.go [hunk 20]
```diff
@@ -0,0 +1,178 @@
...
+		return true
+	}
+	return false
+}
+
+func buildSequenceInfo(stmt *ast.CreateSequenceStmt, ident ast.Ident) (*model.SequenceInfo, error) {
+	sequenceInfo := &model.SequenceInfo{
+		Cache:      model.DefaultSequenceCacheBool,
+		Cycle:      model.DefaultSequenceCycleBool,
+		Order:      model.DefaultSequenceOrderBool,
```
Comment (wjhuang2016) [inline ddl/sequence.go]: Remove the `Order`

File: ddl/sequence.go [hunk 21]
```diff
@@ -0,0 +1,178 @@
...
+		CacheValue: model.DefaultSequenceCacheValue,
+		Increment:  model.DefaultSequenceIncrementValue,
+	}
+
+	// Handle table comment options.
+	for _, op := range stmt.TblOptions {
+		switch op.Tp {
+		case ast.TableOptionComment:
```
Comment (wjhuang2016) [inline ddl/sequence.go]: I can't sure if we need to support Comment.

Comment (wjhuang2016) [reply ddl/sequence.go]: If we really plan to support comment, we have better use single `Comment` instead of `TableOption`

Comment (AilinKid) [reply ddl/sequence.go]: This has been discussed in parser, actually its a option list. Take `comment xxx comment xxx` into consideration, only the last comment counts like collate and charset does.

File: ddl/sequence.go (hunk 10 above)
Comment (bb7133) [reply ddl/sequence.go]: I can keep it so that `handleSequenceOptions` can be used by `ALTER SEQUENCE`.

File: ddl/sequence.go [hunk 22]
```diff
@@ -0,0 +1,176 @@
...
+			sequenceInfo.CacheValue = op.IntValue
+		case ast.SequenceNoCache:
+			sequenceInfo.Cache = false
+		case ast.SequenceCycle:
+			sequenceInfo.Cycle = true
+		case ast.SequenceNoCycle:
+			sequenceInfo.Cycle = false
+		}
+	}
+	// Fill the default value, min/max/start should be adjusted with increment's positive and negative.
```
Comment (bb7133) [inline ddl/sequence.go]:
```suggestion
	// Fill the default value, min/max/start should be adjusted with the sign of sequenceInfo.Increment
```
"positive" and "negative" are both adjectives, they should not be used as objects.

File: ddl/ddl_api.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (AilinKid) [reply ddl/ddl_api.go]: addressed.

File: ddl/sequence.go [hunk 23]
```diff
@@ -0,0 +1,176 @@
...
+				sequenceInfo.Start = mathutil.MinInt64(sequenceInfo.MaxValue, model.DefaultNegativeSequenceStartValue)
+			}
+			if !minSetFlag {
+				sequenceInfo.MaxValue = model.DefaultNegativeSequenceMinValue
+			}
+		}
+	}
+}
+
+func validSequenceOptions(seqInfo *model.SequenceInfo) bool {
```
Comment (bb7133) [inline ddl/sequence.go]: "valid" is an adjective, please rename to "validate..." or "check..."

File: ddl/sequence.go [hunk 24]
```diff
@@ -0,0 +1,176 @@
...
+	// To ensure that cache * increment will never overflows.
+	var maxIncrement int64
+	if seqInfo.Increment != 0 {
+		maxIncrement = math2.Abs(seqInfo.Increment)
+	} else {
+		maxIncrement = math.MaxInt16
+	}
+	if seqInfo.MaxValue >= seqInfo.Start &&
```
Comment (bb7133) [inline ddl/sequence.go]: How about:

```
return seqInfo.MaxValue...
```

File: ddl/sequence.go [hunk 25]
```diff
@@ -0,0 +1,176 @@
...
+	for _, op := range stmt.TblOptions {
+		switch op.Tp {
+		case ast.TableOptionComment:
+			sequenceInfo.Comment = op.StrValue
+		default:
+			return nil, ErrSequenceUnsupportedTableOption.GenWithStackByArgs(op.StrValue)
+		}
+	}
+	handleSequenceOptions(stmt.SeqOptions, sequenceInfo)
+	// Validate the sequence value.
```
Comment (bb7133) [inline ddl/sequence.go]: This comment is redundant(the method name "valid..." is clear enough), please remove it.

File: meta/meta.go [hunk 26]
```diff
@@ -207,6 +212,26 @@ func (m *Meta) GetAutoRandomID(dbID int64, tableID int64) (int64, error) {
 	return m.txn.HGetInt64(m.dbKey(dbID), m.autoRandomTableIDKey(tableID))
 }
 
+// GenSequenceID adds step to the sequence value and returns the sum.
+func (m *Meta) GenSequenceID(dbID, sequenceID, step int64) (int64, error) {
```
Comment (bb7133) [inline meta/meta.go]:
```suggestion
func (m *Meta) GenSequenceValue(dbID, sequenceID, step int64) (int64, error) {
```
How about renaming it to `GenSequenceValue`? The name is a little confusing: we pass a parameter `sequenceID` but the function name is `GenSequenceID`.

I know that we're facing a similar problem with the existing `GenAutoTableID`, but I still think renaming the function here is better.

File: meta/meta.go [hunk 27]
```diff
@@ -207,6 +212,26 @@ func (m *Meta) GetAutoRandomID(dbID int64, tableID int64) (int64, error) {
...
+	// Check if sequence exists.
+	tableKey := m.tableKey(sequenceID)
+	if err := m.checkTableExists(dbKey, tableKey); err != nil {
+		return 0, errors.Trace(err)
+	}
+	return m.txn.HInc(dbKey, m.sequenceKey(sequenceID), step)
+}
+
+// GetSequenceID gets current sequence value with sequence id.
+func (m *Meta) GetSequenceID(dbID int64, sequenceID int64) (int64, error) {
```
Comment (bb7133) [inline meta/meta.go]: Ditto for the function name.

File: ddl/sequence.go (hunk 24 above)
Comment (AilinKid) [reply ddl/sequence.go]: nice idea.

File: meta/meta.go (hunk 26 above)
Comment (AilinKid) [reply meta/meta.go]: addressed

**Pull request merged**